test/e2e/apimachinery/webhook.go

```diff
@@ -703,8 +704,381 @@ var _ = SIGDescribe("AdmissionWebhook [Privileged:ClusterAdmin]", func() {
+		framework.ExpectError(err, "create mutatingwebhookconfiguration should have been denied by the api-server")
+		expectedErrMsg := "compilation failed"
+		if !strings.Contains(err.Error(), expectedErrMsg) {
+			framework.Failf("expect error contains %q, got %q", expectedErrMsg, err.Error())
+		}
+	})
+
+	/*
+		Release: v1.27
+		Testname: Admission webhook, validating webhook exclude leases using match conditions field.
+		Description: Create a validating webhook configuration with matchConditions field that
+		will reject all resources except the coordination.k8s.io/lease ones. Try to create pods
+		until the webhook is ready and rejecting the pods with "denied" error message. Create
+		a Lease object and validate that it bypasses the webhook. Create a configMap and validate
+		that it's rejected by the webhook.
+	*/
+	ginkgo.It("should reject everything except leases [Alpha][Feature:AdmissionWebhookMatchConditions]", func(ctx context.Context) {
+		onlyAllowLeaseObjectMatchConditions := []admissionregistrationv1.MatchCondition{
+			{
+				Name:       "exclude-leases",
+				Expression: `!(request.resource.group == "coordination.k8s.io" && request.resource.resource == "leases")`,
+			},
+		}
+
+		ginkgo.By("creating a validating webhook with match conditions")
+		validatingWebhookConfiguration := newValidatingWebhookWithMatchConditions(f, servicePort, certCtx, onlyAllowLeaseObjectMatchConditions)
+		_, err := createValidatingWebhookConfiguration(ctx, f, validatingWebhookConfiguration)
+		framework.ExpectNoError(err, "registering webhook config %s", f.UniqueName)
+		defer func() {
+			err := client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Delete(ctx, validatingWebhookConfiguration.Name, metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "Deleting mutating webhook configuration")
+		}()
+
+		err = waitWebhookConfigurationReady(ctx, f, f.Namespace.Name)
+		framework.ExpectNoError(err, "waiting for webhook configuration to be ready")
+
+		ginkgo.By("validate that the validating webhook configuration does not reject leases")
+		_, err = client.CoordinationV1().Leases(f.Namespace.Name).Create(ctx, &coordinationv1.Lease{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "lease" + f.UniqueName,
+			},
+			Spec: coordinationv1.LeaseSpec{},
+		}, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "creating coordination.k8s.io/lease object")
+
+		ginkgo.By("validate that the validating webhook configuration does reject configMaps")
+		_, err = client.CoreV1().ConfigMaps(f.Namespace.Name).Create(ctx, &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "cm" + f.UniqueName,
+			},
+			Data: map[string]string{
+				"reject": "this",
+			},
+		}, metav1.CreateOptions{})
+		framework.ExpectError(err, "creating configmap object")
+		expectedErrMsg := "denied"
+		if !strings.Contains(err.Error(), expectedErrMsg) {
+			framework.Failf("expect error contains %q, got %q", expectedErrMsg, err.Error())
```

Comment:
This should be framework.Errorf since we are failing an assertion and not a setup step

Reply:
`framework` and `ginkgo` don't have an `Errorf` method - i see `Failf` is used in other places for e2e tests.

Reply:
The ginkgo way of doing this would be:
```
framework.Expect(strings.Contains(err.Error(), expectedErrMsg)).To(BeTrue())  
```

Or something like this